sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTest.java

```diff
@@ -163,6 +179,55 @@ public void processElement(ProcessContext c) {
     readAfterWritePipeline.run();
   }
 
+  private Map<String, String> getPartitions() {
+    Map<String, String> partitions = new HashMap<>();
+    partitions.put("load_date", "2019-05-14T23:28:04.425Z");
+    partitions.put("product_type", "1");
+    return partitions;
+  }
+
+  /** Perform end-to-end test of Write-then-Read operation. */
+  @Test
+  @NeedsEmptyTestTablesForUnboundedReads
+  public void testWriteThenUnboundedReadSuccess() throws Exception {
+
+    defaultPipeline
+        .apply(Create.of(buildHCatRecords(TEST_RECORDS_COUNT)))
+        .apply(
+            HCatalogIO.write()
+                .withConfigProperties(getConfigPropertiesAsMap(service.getHiveConf()))
+                .withDatabase(TEST_DATABASE)
+                .withTable(TEST_TABLE)
+                .withPartition(getPartitions())
+                .withBatchSize(512L));
+    defaultPipeline.run();
+    final ImmutableList<String> partitions = ImmutableList.of("load_date", "product_type");
+    final PCollection<HCatRecord> data =
+        readAfterWritePipeline
+            .apply(
+                "ReadData",
+                HCatalogIO.read()
+                    .withConfigProperties(getConfigPropertiesAsMap(service.getHiveConf()))
+                    .withDatabase(TEST_DATABASE)
+                    .withPartitionCols(partitions)
+                    .withTable(TEST_TABLE)
+                    .withPollingInterval(Duration.millis(15000)))
```

Comment:
Does this test terminate? I think your code sets up Watch.growthOf() in a way that doesn't terminate..

Reply:
It didn't earlier. Added the ability to specify a termination condition